src/TestInfrastructure/Helpers/ProcessCleanupHelper.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Helpers
+{
+    /// <summary>
+    /// Tries to ensure the process is terminated when a test finishes.
+    /// Usage:
+    ///   using var cleanHelper = new ProcessCleanupHelper(process);
+    /// </summary>
+    /// <remarks>Disposing a process does not terminate it, so we can't rely on
+    /// "using var process = ..." to clean up properly.
+    /// Instead, we'll try to fetch the process by id then "Kill" it.
+    /// </remarks>
```

Comment:
Originally, I was using the 'using [a Process instance]`, assuming that the `Process.Dispose()` would terminate the process, but it doesn't.

However, `using var cleanupHelper ...` will.